### PR TITLE
MAINT: Update PCG jump sizes

### PR DESCRIPTION
**Pull request by bashtage**
Update pcg jump sizes
Wrap advance when too large

### CODE REVIEW

**Comment by mattip** [comment]
pushed a commit to remove some trailing spaces

File: numpy/random/pcg32.pyx [hunk 1]
```diff
@@ -248,24 +248,29 @@ cdef class PCG32:
     cdef jump_inplace(self, iter):
         """
         Jump state in-place
-
+        
         Not part of public API
-
+
```
Comment (mattip) [inline numpy/random/pcg32.pyx]: These blank lines have trailing spaces

File: numpy/random/common.pyx [hunk 2]
```diff
@@ -182,6 +182,16 @@ cdef double kahan_sum(double *darr, np.npy_intp n):
...
     return sum
 
+
+cdef object wrap_int(object val, object bits):
+    """Wraparound to place an integer into the interval [0, 2**bits)"""
+    upper = int(2)**int(bits)
+    if not 0<= val < upper:
+        divisor = val // upper
+        val = val - upper * divisor
+    return val
```
Comment (eric-wieser) [inline numpy/random/common.pyx]: Would this be faster as follows?
```
mask = ~(~int(0) << bits)
return val & mask
```

File: numpy/random/pcg32.pyx (hunk 1 above)
Comment (bashtage) [reply numpy/random/pcg32.pyx]: Removed, and a couple of bonus trailing whitespaces

File: numpy/random/common.pyx (hunk 2 above)
Comment (bashtage) [reply numpy/random/common.pyx]: Thanks - seems to work identically.

File: numpy/random/pcg32.pyx [hunk 3]
```diff
@@ -255,17 +255,22 @@ cdef class PCG32:
...
 
     def jumped(self, iter=1):
         """
         jumped(iter=1)
 
         Returns a new bit generator with the state jumped
 
-        The state of the returned big generator is jumped as-if
-        2**(32 * iter) random numbers have been generated.
+        Jumps the state as-if 11400714819323198486 random numbers
```
Comment (eric-wieser) [inline numpy/random/pcg32.pyx]: Shouldn't this involve `iter` in the description?

File: numpy/random/pcg64.pyx [hunk 4]
```diff
@@ -264,17 +264,26 @@ cdef class PCG64:
...
+        Notes
+        -----
+        The step size is phi-1 when divided by 2**128 where phi is the
+        golden number.
         """
-        self.advance(iter * 2**64)
+        step = 0x9e3779b97f4a7c15f39cc0605cedc834
+        step *= int(iter)
+        divisor = step // 2**128
+        step -= 2**128 * divisor
```
Comment (eric-wieser) [inline numpy/random/pcg64.pyx]: Isn't this `step = wrap_int(step, 128)`?

**Pull request merged**